src/mono/wasm/build/WasmApp.targets

```diff
@@ -5,11 +5,10 @@
 
   <!--
       Required public items/properties:
-      - $(WasmMainJSPath)
-
       - $(EMSDK_PATH)      - points to the emscripten sdk location.
 
       Public properties (optional):
+      - $(WasmMainJSPath)    - Main JavaScript for execution on V8/node
```

Comment:
There is a mismatch here. This change suggests that we want to make `$(WasmMainJSPath)` an optional property. But changes in the rest of the PR - namely setting the value in the projects, and then explicitly adding it to `@(WasmExtraFilesToDeploy)` suggests that there isn't any handling of this property in the core targets anymore.

Maybe the intention is to:

1. Remove the mainjs, and html stuff from WasmAppBuilder (as is done here)
2. Automatically add `$(WasmMainJSPath)` to `WasmExtraFilesToDeploy` in WasmApp.targets, if set

That would essentially keep `$(WasmMainJSPath)` as a property known to WasmApp.targets, but would not require it anymore.

Reply:
The property is used in the `_GenerateRunV8Script` target. It's the only place.

Reply:
V8 script generation is hidden behind `$(WasmGenerateRunV8Script) == true` and the `WasmGenerateRunV8Script` is not documented in the `WasmApp.targets`. So I have dropped the comment for `$(WasmMainJSPath)`.

Reply:
At least for back compat we should add `$(WasmMainJSPath)` to `@(WasmExtraFilesToDeploy)` if set. Maybe I'm missing a reason for avoiding this?

Reply:
`WasmMainJSPath` must not be mandatory because sometimes we don't own the entrypoint, we are just JS+wasm library.
Making if backward compatible with old templates makes sense to me.